GitUI/CommandsDialogs/SettingsDialog/Pages/ChecklistSettingsPage.cs

```diff
@@ -587,59 +502,55 @@ private bool CheckGitCmdValid()
         private bool CheckDiffToolConfiguration()
         {
             DiffTool.Visible = true;
-            if (string.IsNullOrEmpty(CommonLogic.GetGlobalDiffTool()))
+            string difftool = CommonLogic.GetGlobalDiffTool();
+            if (string.IsNullOrEmpty(difftool))
             {
                 RenderSettingUnset(DiffTool, DiffTool_Fix, _adviceDiffToolConfiguration.Text);
                 return false;
             }
 
-            if (EnvUtils.RunningOnWindows())
+            string cmd = GetGlobalSetting($"difftool.{difftool}.cmd");
+            if (cmd.IsNullOrWhiteSpace())
             {
-                if (CommonLogic.IsDiffTool("kdiff3"))
+                cmd = MergeToolsHelper.DiffToolCmdSuggest(difftool, "");
+                if (cmd.IsNullOrWhiteSpace())
                 {
-                    string p = GetGlobalSetting("difftool.kdiff3.path");
-                    if (string.IsNullOrEmpty(p) || !File.Exists(p))
-                    {
-                        RenderSettingUnset(DiffTool, DiffTool_Fix, _kdiffAsDiffConfiguredButNotFound.Text);
-                        return false;
-                    }
-
-                    RenderSettingSet(DiffTool, DiffTool_Fix, _kdiffAsDiffConfigured.Text);
-                    return true;
+                    RenderSettingUnset(DiffTool, DiffTool_Fix, _adviceDiffToolConfiguration.Text);
+                    return false;
                 }
             }
 
-            string difftool = CommonLogic.GetGlobalDiffTool().ToLowerInvariant();
             RenderSettingSet(DiffTool, DiffTool_Fix, string.Format(_diffToolXConfigured.Text, difftool));
             return true;
         }
 
         private bool CheckMergeTool()
         {
             MergeTool.Visible = true;
-            if (string.IsNullOrEmpty(CommonLogic.GetGlobalMergeTool()))
+            string mergetool = CommonLogic.GetGlobalMergeTool();
+            if (string.IsNullOrEmpty(mergetool))
             {
                 RenderSettingUnset(MergeTool, MergeTool_Fix, _configureMergeTool.Text);
                 return false;
             }
 
-            string mergetool = CommonLogic.GetGlobalMergeTool().ToLowerInvariant();
+            // Hardcode parameters to some mergetools
             if (EnvUtils.RunningOnWindows())
             {
-                if (CommonLogic.IsMergeTool("kdiff3"))
+                if (mergetool.ToLowerInvariant() == "kdiff3")
```

Comment:
I suggest we completely remove all explicit dependencies on specfic tools

Reply:
Yes, this PR is changing the behavior, not auto suggesting KDiff3 (as the exe is not bundled). For KDiff3 GE has knowledge about parameters etc, why it is specially handled here.

Reply:
> We can remove the rest in the follow up work

I have been staring at these lines of code for a long time now and do not fully realize what to replace it with.
The current code is handling the previous but should be removed. A follow up is needed anyway.